crates/adapters/src/format/avro/schema.rs

```diff
@@ -128,6 +128,39 @@ fn validate_map_schema(
     Ok(())
 }
 
+/// Check that Avro schema can be deserialized as SQL `BIGINT` type.
+fn validate_bigint_schema(avro_schema: &AvroSchema) -> Result<(), String> {
+    match avro_schema {
+        AvroSchema::Long | AvroSchema::Int => Ok(()),
+        _ => Err(format!(
+            "invalid Avro schema for a column of type 'BIGINT': expected 'long' or 'int', but found {}",
+            schema_json(avro_schema)
+        )),
+    }
+}
+
+/// Check that Avro schema can be deserialized as SQL `INT UNSIGNED` type.
+fn validate_uint_schema(avro_schema: &AvroSchema) -> Result<(), String> {
+    match avro_schema {
+        AvroSchema::Long | AvroSchema::Int => Ok(()),
+        _ => Err(format!(
+            "invalid Avro schema for a column of type 'INT UNSIGNED': expected 'long' or 'int', but found {}",
+            schema_json(avro_schema)
+        )),
+    }
+}
+
+/// Check that Avro schema can be deserialized as SQL `BIGINT UNSIGNED` type.
+fn validate_ubigint_schema(avro_schema: &AvroSchema) -> Result<(), String> {
+    match avro_schema {
+        AvroSchema::Long | AvroSchema::Int | AvroSchema::String => Ok(()),
```

Comment:
why string only here?
you'd think it should be allowed everywhere.
What happens if the values are out of range (for all types)? Is there an additional runtime check?